workshops/boost-research-reproducibility-binder/workshop-presentations/zero-to-binder.md

```diff
@@ -0,0 +1,254 @@
+
+#### Large public files
+
+It is not practical to place large files in your GitHub repo or include them directly in the image that Binder builds.
+The best option for large files is to use a library specific to the data format to stream the data as you're using it or to download it on demand as part of your code.
+
+For security reasons, the outgoing traffic of your Binder is restricted to HTTP or GitHub connections only. You will not be able to use FTP sites to fetch data on mybinder.org.
+
+#### Private files
+
+There is no way to access files which are not public from mybinder.org.
+You should consider all information in your Binder as public, meaning that:
+* there should be no passwords, tokens, keys etc in your GitHub repo;
+* you should not type passwords into a Binder running on mybinder.org;
+* you should not upload your private SSH key or API token to a running Binder.
+
+In order to support access to private files, you would need to create a local deployment of [BinderHub](https://binderhub.readthedocs.io/en/latest/) where you can decide the security trade offs yourselves.
+
+## 8. Get data with `postBuild`
+
+**TO DO:** :vertical_traffic_light:
+
+1) Go to your GitHub repo and create a file called `postBuild`
+2) In `postBuild`, add a single line reading: `wget -q -O gapminder.csv http://bit.ly/2uh4s3g`
+3) Update your `requirements.txt` file by adding a new line with `pandas` on it and another new line with `matplotlib` on it
+   * These packages aren't necessary to download the data but we will use them to read the CSV file and make a plot
+4) Click the binder badge in your README to launch your Binder
+
+Once the Binder has launched, you should see a new file has appeared that was not part of your repo when you clicked the badge.
+
+Now visualise the data by creating a new notebook ("New" :arrow_right: "Python 3") and run the following code in a cell.
+
+```python
+%matplotlib inline
+
+import pandas
+
+data = pandas.read_csv("gapminder.csv", index_col="country")
+
+years = data.columns.str.strip("gdpPercap_")  # Extract year from last 4 characters of each column name
+data.columns = years.astype(int)              # Convert year values to integers, saving results back to dataframe
+
+data.loc["Australia"].plot()
+```
+
+See this [Software Carpentry lesson](https://swcarpentry.github.io/python-novice-gapminder/09-plotting/index.html) for more info.
+
+## Beyond Notebooks...
+
+**JupyterLab** is installed into your containerized repo by default.
+You can access the environment by changing the URL you visit to:
+
+> **https://mybinder.org/v2/gh/your-username/my-first-binder/master?urlpath=lab**
+
+**N.B.:** We've already seen how the `?filepath=` argument can link to a specific file.
+
+Here you can access:
+* Notebooks
+* IPython consoles
+* Terminals
+* A text editor
+
+If you use R, you can also open **RStudio** using `?urlpath=rstudio`.
+
+## Now over to you!
```

Comment:
More of a point for your talk rather than the content of this file, but coffee should arrive from 2pm so this might be a natural point for a break - if anyone starts reading further down the doc they could end up getting ahead of the rest of the group.

Reply:
Yeah, this is totally designed as a break point. They'll either get here before 2pm and they can start playing early, or we break for coffee and then I put this info back up for them.